src/core/MythosArea.ttslua

```diff
@@ -201,6 +201,27 @@ function reshuffleEncounterDeck(params)
   Wait.time(function() isReshuffling = false end, 1)
 end
 
+function discardIntoDeck()
+  --flag to avoid multiple calls
+  if isReshuffling then return end
+  isReshuffling = true
+
+  -- flip deck and then shuffle
```

Comment:
You can just combine the deck objects directly using the `putObject` [function](https://api.tabletopsimulator.com/object/#putobject).  It should drastically simplify the logic here.

Reply:
Yeah, @Chr1Z93 has already mentioned this. I'm going to work on this a bit more and clean up the other encounter draw functions in the same script so they all match.

Re: nicheness -- the other use case is when the encounter deck runs out on its own. It only reshuffles when you draw an encounter card, but the encounter deck has to reshuffle after the last card is drawn and resolved. So, e.g. if the last card drawn is an enemy, if you defeat it, it goes into the discard after the encounter deck has already reshuffled. If there's another spot for it, I'm all ears, but IMO it's an important enough function to not just be a hotkey.

Reply:
Oh wow that's super interesting, I didn't know that's how the rules worked!  Definitely been playing that one wrong XD. 
 Too much muscle memory from other card games (even this one!  Horror/reshuffle of player cards only triggers on the next instance of draw, not when empty.  What an odd deviation).  In this case, I think it would be better to just change the existing script to check if the encounter deck is empty _after_ drawing, rather than before, and trigger the reshuffle, in accordance to the rules, as normal there.  

Then that leaves this button with just the first niche use case, which I don't think justifies its existence (as a button, hotkey or anything else).  We have very specifically strayed away from automating scenario-specific effects (aside from the odd set-up helper), so I don't really see a reason to start now, especially for such a basic action.

Reply:
I may have communicated this poorly -- the encounter card draw script definitely should stay as it is -- we can't predict when a card has finished resolving, so the encounter deck reshuffle has to be triggered manually.

Reply:
Aaaaaah I see what you're saying now.  Hmmm I'm starting to come around to the use case then.  I still think we want to add 2 broadcast messages then: one when the deck is empty, to remind people to click the button to reshuffle, and one if the old reshuffle script is triggered, to 'chastise' them for it a bit and make sure they're playing it correctly.  Or maybe even disable the default reshuffle altogether honestly, to avoid that play pattern trap.

That being said, if it is going to be much more frequently called, I would consider also making a hotkey for it.  I'd let @Chr1Z93 weigh in on that decision though.

Reply:
I'm also rewriting some of the rest of the MythosArea script under @Chr1Z93's tutelage -- some of this script makes parts of the other scripts redundant, whether or not we have a button to call it. You'll see some more soon :-D